core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -656,7 +656,8 @@ public NativeImageInvokerInfo build() {
                 if (!protocols.isEmpty()) {
                     nativeImageArgs.add("-H:EnableURLProtocols=" + String.join(",", protocols));
                 }
-                if (enableAllSecurityServices) {
+                if (enableAllSecurityServices && graalVMVersion.isOlderThan(GraalVM.Version.VERSION_21_1)) {
```

Comment:
Should we log a warning if the flag is not going to be applied?

Reply:
I thought about it and decided not to since this flag cannot be deprecated yet (since it's still needed by GraalVM 20.3).
Printing a warning here would just confuse users IMO.

Instead I think it's OK to silently skip it in GraalVM >= 21.1 and once we drop support for 20.3 we can deprecate the option and print a warning.